Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/remesh_planar_patches.h

```diff
@@ -0,0 +1,1649 @@
+        for (std::size_t cid=ncid; cid<corners.size(); ++cid)
+        {
+          typedef std::pair<const std::size_t, vertex_descriptor> Map_pair_type;
+          auto find_res = point_to_vertex_maps.find(corners[cid]);
+          assert(find_res != point_to_vertex_maps.end());
+          for(Map_pair_type& mp : find_res->second)
+          {
+            std::size_t other_mesh_id = mp.first;
+            if ( other_mesh_id!=mesh_id  && !is_corner_id(get(vertex_corner_id_maps[other_mesh_id], mp.second)))
+            {
+              mesh_ids.insert(other_mesh_id);
+              put(vertex_corner_id_maps[other_mesh_id], mp.second,
+                nb_corners_and_nb_cc_all[other_mesh_id].first++);
+              if (!all_corners[other_mesh_id].empty())
+                all_corners[other_mesh_id].push_back(corners[cid]);
+            }
+          }
+        }
+        for (std::size_t mid : mesh_ids)
+          if (!to_be_processed[mid])
+          {
+#ifdef CGAL_DEBUG_DECIMATION
+            if (!loop_again)
+              std::cout << "setting for another loop\n";
+#endif
+            loop_again=true;
+            to_be_processed[mid] = true;
+          }
+      }
+      to_be_processed[mesh_id] = false;
+    }
+  }
+  while(loop_again);
+
+  // now create the new meshes:
+  for(std::size_t mesh_id=0; mesh_id<nb_meshes; ++mesh_id)
+  {
+    Triangle_mesh& tm = *mesh_ptrs[mesh_id];
+    if (mesh_has_non_manifold_vertices[mesh_id])
+    {
+//      no_remeshing_issue = false;
+      continue;
+    }
+    CGAL_assertion(is_polygon_soup_a_polygon_mesh(all_faces[mesh_id]));
+
+    //clear(tm);
+    tm.clear_without_removing_property_maps();
+    polygon_soup_to_polygon_mesh(all_corners[mesh_id], all_faces[mesh_id],
+                                 tm, parameters::default_values(), parameters::vertex_point_map(vpms[mesh_id]));
+  }
+
+  return res;
+}
+
+} //end of namespace Planar_segmentation
+
+
+/*!
+ *  \ingroup PMP_meshing_grp
+ *  generates a new triangle mesh `tm_out` with the minimal number of triangles while preserving the shape as `tm_in`.
+ *  In practice, this means that connected components of edge incident faces belonging to the same plane are
+ *  first extracted (each such connected component is called a *patch*). Then the connected components of vertex
+ *  connected patch border edges belonging to the same line are extracted. Endpoints of such components and
+ *  vertices incident to more than two patches (or two patches + one mesh boundary) are called *corners*.
+ *  `tm_out` contains the 2D constrained Delaunay triangulation of each patch using only corner vertices
+ *  on the boundary of the patch.
+ *
+ *  \warning if `tm_in` contains a non-manifold vertex, `tm_out` will be empty. Those vertices must be
+ *           duplicated with `duplicate_non_manifold_vertices()` to get an output.
+ *
+ *  \tparam TriangleMeshIn a model of `HalfedgeListGraph` and `FaceListGraph`
+ *  \tparam TriangleMeshOut a model of `MutableFaceGraph`
```

Comment:
It's not a triangle mesh if you don't triangulate

also " *  \param tm_out output triangle mesh"

Reply:
My reasoning is: "Polygonal is more generic so it is a stronger requirement than triangle mesh. Thus, if sometimes you need polygonal, it should be polygonal that is documented."